Source/Bogus/Randomizer.cs

```
@@ -75,46 +75,79 @@ public int[] Digits(int count, int minDigit = 0, int maxDigit = 9)
+            // number from a range spanning all possible values of int. The Random class only supports exclusive
+            // upper bounds, period, and the upper bound must be specified as an int, so the best we can get in a
+            // single call is a value in the range (int.MinValue, int.MaxValue - 1). Instead, what we do is get two
+            // samples, one in the range (int.MinValue, -1) and the other as unbiased as possible, and using the
+            // second one to decide, 50% of the time we invert all the bits in the sample, shifting its range to
+            // (0, int.MaxValue).
+            var result = localSeed.Next(int.MinValue, 0);
+
+            if( (localSeed.Next() & 0x10000000) == 0 )
+               result = ~result;
+
+            return result;
          }
       }
 
       /// <summary>
-      /// Returns a random even number.
+      /// Returns a random even number. If the range does not contain any even numbers, the first even number after the range is returned, or int.MinValue if min is int.MaxValue.
       /// </summary>
       /// <param name="min">Lower bound, inclusive</param>
       /// <param name="max">Upper bound, inclusive</param>
       public int Even(int min = 0, int max = 1)
       {
-         var result = 0;
-         do //could do this better by just +1 or -1 if it's not an even/odd number
-         {
-            result = Number(min, max);
-         } while( result % 2 == 1 );
-         return result;
+         // Special case where the math below breaks.
+         if( min == int.MaxValue )
+            return int.MinValue;
+
+         // Adjust the range to ensure that we always get the same number of even values as odd values.
+         min = (min + 1) & ~1;
+         max = max | 1;
+
+         if( min > max )
+            return min;
+
+         // Strip off the 1 bit of a random number.
+         return Number(min, max) & ~1;
       }
```

I'm thinking we should probably throw an exception if the user gets it wrong.

For example, if the user calls `.Even(min:3, max:3)` there are no even numbers; however, in this change, we would return 4 which seems unsuspecting. The user might not be aware of their mistake.

Kinda feel, if I call `.Even(min, max)` I should get exactly that, between `min` and `max`. Semantics wise, I would not expect a value outside of `min` and `max`; even if I made the mistake on purpose.

I assumed when I wrote many of these `Randomizer` methods that the inputs would be correct and if the user got it wrong, "oh well". I suppose if we're going to get proper about it, I think we can lock down these out of bounds arguments with exception guards.

Let me know your thoughts.

---

LoL, the production version in `master` enters an infinite loop when there are no `.Even` numbers:
```csharp
void Main()
{
   var f = new Faker();
   f.Random.Even(3, 3).Dump();
}
```
So I guess, the current behavior is that we've already broken the user's program and they might have noticed this when Bogus got stuck.